Added Duration datastructure

adognameddog opened the pull request:
```
## Duration (object)
+ unit: `s` (string, required) - Seconds
+ value: `10800` (number, required)
```

adognameddog committed:
:art: Added Duration datastructure

JordyDeGroeve reviewed (dismissed):
Will it be used in the near future?

adognameddog commented:
> Will it be used in the near future?

I don't know yet. Debating wether to use in on `timeTracking.report`, but I don't know if we want to suggest that we might return a duration with, suddenly, a new unit

JordyDeGroeve commented:
Not sure if this should be added if there is no concrete use case for it.
We can leave out the other units perhaps?
What does @stivni think?

adognameddog committed:
:memo: Duration datastructure is always in seconds

Seconds is the only SI unit for a duration of time, and it is also the smallest granularity on which we allow to track time.

adognameddog commented:
> Not sure if this should be added if there is no concrete use case for it.
> We can leave out the other units perhaps?
> What does @stivni think?

I'm going to use it in the docs for `timeTracking.report`

tiagobastos commented on apiary.apib:
```diff
@@ -4307,3 +4307,7 @@ Unregister a webhook.
                 + `payment_card`
+
+## Duration (object)
++ unit: `s` (string, required) - Seconds
```
Q: Why not do it like: 
```
+ unit: `s` (enum[string], required)
    + Members
        + s - Seconds
```

We _"did it"_ for Distance on Work Orders.
https://github.com/teamleadercrm/api-internal/pull/60/files#diff-1841f13b750b4c8d06587142329fe64cR177-R181

joaoinacio replied:
`DurationUnit` also provides minutes, as `min` so that would actually make it:
```
+ unit: `s` (enum[string], required)
    + Members
        + s - Seconds
		+ min - Minutes
```

joaoinacio replied:
We already use duration in minutes on some endpoints, so having this support it would be ideal IMO.

tiagobastos replied:
And not in the scope of this proposal to change, but to provide more insight, the `/tasks.create` already uses a duration like structure on `estimated_duration`. 
That could be a good example of it's usage already.

dietrichm replied:
Agreeing with @joaoinacio and @tiagobastos. Let's define `unit` as an enum so we are not fixed to having just one possible unit.

adognameddog committed:
:ok_hand: Unit is an enumeration

Although, for now, we only return or accept seconds as the unit

ketlane reviewed (commented):
The already existing documentation uses another notation when expressing duration units: `min` for minutes (i.e. on `tasks.list` for `estimated_duration`). That would mean for seconds we'd have to use `sec` not `s`, if we want to be consistent.  Otherwise, changing `min` to `m` in already implemented / documented endpoints would imply a breaking change.

So I'm proposing to use `sec` instead of `s`.

joaoinacio commented:
> The already existing documentation uses another notation when expressing duration units: `min` for minutes (i.e. on `tasks.list` for `estimated_duration`). That would mean for seconds we'd have to use `sec` not `s`, if we want to be consistent. Otherwise, changing `min` to `m` in already implemented / documented endpoints would imply a breaking change.
> 
> So I'm proposing to use `sec` instead of `s`.

However there is also already existing documentation using `s` notation for seconds.
This really should be standardised across existing endpoints, but either way it would/will involve some kind of (breaking?) change.

Also, [DurationUnit](https://github.com/teamleadercrm/core/blob/master/src/Tasks/Application/Http/DurationUnit.php#L17) does use `s` and `min`... 🤷‍♂

tiagobastos commented:
Should we decide on the implementation we would like to see in all the endpoints?
And the ones that differ would be adjusted in the context of each team. 
> So I'm proposing to use sec instead of s.

Like this approach too.

adognameddog commented:
> The already existing documentation uses another notation when expressing duration units: `min` for minutes (i.e. on `tasks.list` for `estimated_duration`). That would mean for seconds we'd have to use `sec` not `s`, if we want to be consistent. Otherwise, changing `min` to `m` in already implemented / documented endpoints would imply a breaking change.
> 
> So I'm proposing to use `sec` instead of `s`.

Absolutely not. The international standard abbreviation for seconds is "s". See: https://maths.mq.edu.au/numeracy/tutorial/siunits.htm

We use "min" for minutes because minutes is not a SI unit, and therefor we can practically chose the abbreviation. "m" is metres.

dietrichm reviewed (approved):
Absolutely in favour of using SI units (`s` for seconds).
